src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -1142,20 +1142,20 @@ internal static async Task<ContainerImagePushOptions> ProcessImagePushOptionsCal
     /// </remarks>
     internal static IContainerRegistry GetContainerRegistry(this IResource resource)
     {
-        // Try to get the container registry from DeploymentTargetAnnotation first
-        var deploymentTarget = resource.GetDeploymentTargetAnnotation();
-        if (deploymentTarget?.ContainerRegistry is not null)
-        {
-            return deploymentTarget.ContainerRegistry;
-        }
-
         // Try ContainerRegistryReferenceAnnotation (explicit WithContainerRegistry call)
         var registryAnnotation = resource.Annotations.OfType<ContainerRegistryReferenceAnnotation>().LastOrDefault();
         if (registryAnnotation is not null)
         {
             return registryAnnotation.Registry;
         }
 
+        // Try to get the container registry from DeploymentTargetAnnotation first
```

Comment:
Is there a test that ensures ContainerRegistryReferenceAnnotation overrides the DeploymentTargetAnnotation? For example with ACA or AppService?